core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -1626,6 +1639,38 @@ private ExprValue exprToDays(ExprValue date) {
     return new ExprLongValue(date.dateValue().toEpochDay() + DAYS_0000_TO_1970);
   }
 
+  /**
+   * To_seconds implementation for ExprValue.
+   *
+   * @param date ExprValue of Date/Datetime/Timestamp/String type.
+   * @return ExprValue.
+   */
+  private ExprValue exprToSeconds(ExprValue date) {
+    return new ExprLongValue(
+        date.datetimeValue().toEpochSecond(ZoneOffset.UTC) + DAYS_0000_TO_1970 * SECONDS_PER_DAY);
+  }
+
+  /**
+   * To_seconds implementation with an integer argument for ExprValue.
+   *
+   * @param dateExpr ExprValue of an Integer/Long formatted for a date (e.g., 950501 = 1995-05-01)
+   * @return ExprValue.
+   */
+  private ExprValue exprToSecondsForIntType(ExprValue dateExpr) {
+    try {
+      if(dateExpr.longValue() < 0 || dateExpr.longValue() > 999999) {
+        throw new DateTimeException("Integer argument was out of range");
+      }
+
+      LocalDate date = LocalDate.parse(String.valueOf(dateExpr.longValue()), DATE_FORMATTER_SHORT_YEAR);
```

Comment:
This may need to be adjusted. MySQL accepts formats `yyyymmdd`, `yymmdd`, `ymmdd`.

Reply:
I have reused the implementation.... I'm not sure I like the implementation though so I may try to rework it.